controllers/statuscheck/http/http.go

```diff
@@ -0,0 +1,125 @@
+// Copyright Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package http
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+)
+
+type httpExecutor struct {
+	logger logr.Logger
+}
+
+func NewExecutor(logger logr.Logger) *httpExecutor {
+	return &httpExecutor{logger: logger}
+}
+
+type response struct {
+	statusCode int
+	body       string
+}
+
+func (e *httpExecutor) Type() string {
+	return "HTTP"
+}
+
+func (e *httpExecutor) Do(spec v1alpha1.StatusCheckSpec) (bool, string, error) {
+	client := &http.Client{
```

Comment:
Just a notice: the default `http.Client` will follow the redirect (for at most 10 times), is it expected?

Then the `30x` in the criteria is meaningless.

Reply:
I think we could add this notice to documents later 🤔